internal/gatewayapi/extensionserverpolicy.go

```diff
@@ -0,0 +1,202 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package gatewayapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	gwv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gwv1b1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/envoyproxy/gateway/internal/gatewayapi/status"
+	"github.com/envoyproxy/gateway/internal/ir"
+	"github.com/envoyproxy/gateway/internal/utils"
+)
+
+func (t *Translator) ProcessExtensionServerPolicies(policies []unstructured.Unstructured,
+	gateways []*GatewayContext,
+	xdsIR XdsIRMap,
+) []unstructured.Unstructured {
+	var res []unstructured.Unstructured
+
+	// Sort based on timestamp
+	sort.Slice(policies, func(i, j int) bool {
+		iTime := policies[i].GetCreationTimestamp()
+		jTime := policies[j].GetCreationTimestamp()
+		return iTime.Before(&jTime)
+	})
+
+	// First build a map out of the gateways for faster lookup
+	gatewayMap := map[types.NamespacedName]*policyGatewayTargetContext{}
+	for _, gw := range gateways {
+		key := utils.NamespacedName(gw)
+		gatewayMap[key] = &policyGatewayTargetContext{GatewayContext: gw}
+	}
+
+	// Process the policies targeting Gateways. Only update the policy status if it was accepted.
+	// A policy is considered accepted if at least one targetRef contained inside matched a listener.
+	for _, policy := range policies {
+		policy := policy.DeepCopy()
+		var policyStatus gwv1a2.PolicyStatus
+		res = append(res, *policy)
+		targetRefs, err := extractTargetRefs(policy)
+		if err != nil {
+			t.Logger.Sugar().Errorf("error finding targetRefs for policy %s: %w", policy.GetName(), err)
+			continue
+		}
+		for _, currTarget := range targetRefs {
+			if currTarget.Kind != KindGateway {
+				// TODO: mark this targetRef as an error
+				t.Logger.Sugar().Errorf("extension policy %s doesn't target a Gateway", policy.GetName())
```

Comment:
we've tried to keep the gateway-api lib as minimal as possible, so it can be reused again in other places in the future, can we instead collect `errs` here and return and let the runner log them ?

Reply:
Done.